### PR TITLE
Add support for Drupal 8.7

### PR DESCRIPTION
**Pull request by MPParsley**
## Fixes
- Deprecated EntityTypeManager #26

### CODE REVIEW

File: og_sm_content/src/Plugin/EntityReferenceSelection/NodeSelection.php [hunk 1]
```diff
@@ -48,7 +48,7 @@ class NodeSelection extends NodeSelectionBase {
    * @param \Drupal\og_sm\SiteManagerInterface $site_manager
    *   The site manager.
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityManagerInterface $entity_manager, ModuleHandlerInterface $module_handler, AccountInterface $current_user, SiteManagerInterface $site_manager) {
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, ModuleHandlerInterface $module_handler, AccountInterface $current_user, SiteManagerInterface $site_manager) {
     parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_manager, $module_handler, $current_user);
```
Comment (MPParsley) [inline og_sm_content/src/Plugin/EntityReferenceSelection/NodeSelection.php]:
```suggestion
    parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager, $module_handler, $current_user);
```

**Review by zero2one — APPROVED** [review]

**Review by wesleydv — APPROVED** [review]

**Comment by sandipt27** [comment]
Thanks @MPParsley, However, Changes are still failing.  

You missed to update line #50 in `src/Theme/AdminNegotiator.php` from `parent::__construct($user, $config_factory, $entity_manager, $admin_context);` to `parent::__construct($user, $config_factory, $entity_type_manager, $admin_context);`.  The third parameter to be specific need to be referenced with entity_type_manager instead of entity_manager.

Once that's changed it would fix it.  

I myself would have provided the patch for this one in this same branch but I am a little unsure how to I go for it.  If you could help me understand how to do it, would appreciate a-lot.  Do I need to clone the branch, make a necessary changes and push that complete branch  OR do I need to provide just a patch for specific change ?  AND do I need some additional rights to it ?  Any help to understand it would be highly appreciated.

Thanks

File: src/Theme/AdminNegotiator.php [hunk 2]
```diff
@@ -46,7 +46,7 @@ class AdminNegotiator extends AdminNegotiatorBase {
    * @param \Drupal\og\OgAccessInterface $og_access
    *   The og access service.
    */
-  public function __construct(AccountInterface $user, ConfigFactoryInterface $config_factory, EntityManagerInterface $entity_manager, AdminContext $admin_context, SiteManagerInterface $site_manager, OgAccessInterface $og_access) {
+  public function __construct(AccountInterface $user, ConfigFactoryInterface $config_factory, EntityTypeManagerInterface $entity_type_manager, AdminContext $admin_context, SiteManagerInterface $site_manager, OgAccessInterface $og_access) {
     parent::__construct($user, $config_factory, $entity_manager, $admin_context);
```
Comment (MPParsley) [inline src/Theme/AdminNegotiator.php]: Add missing $entity_type_manager:
```suggestion
    parent::__construct($user, $config_factory, $entity_type_manager, $admin_context);
```
@sandipt27, note that github allows you to make inline suggestions (See number 6 on https://help.github.com/en/articles/commenting-on-a-pull-request).

Also have a look at https://help.github.com/en/articles/creating-a-pull-request-from-a-fork

So if you like to contribute, fork this project and create pull requests to suggest changes.

Good luck!

**Comment by MPParsley** [comment]
> Thanks @MPParsley, However, Changes are still failing.

Tests are failing because we need to update Travis to support the latest versions.

> You missed to update line #50 in `src/Theme/AdminNegotiator.php` from `parent::__construct($user, $config_factory, $entity_manager, $admin_context);` to `parent::__construct($user, $config_factory, $entity_type_manager, $admin_context);`. The third parameter to be specific need to be referenced with entity_type_manager instead of entity_manager.

Thanks, I fixed that. Note that this type of one liners can easily be suggested within a comment using the suggestion button.

> 
> Once that's changed it would fix it.
> 
> I myself would have provided the patch for this one in this same branch but I am a little unsure how to I go for it. If you could help me understand how to do it, would appreciate a-lot. Do I need to clone the branch, make a necessary changes and push that complete branch OR do I need to provide just a patch for specific change ? AND do I need some additional rights to it ? Any help to understand it would be highly appreciated.
> 
> Thanks

If you would like to contribute, fork this project and create pull requests to suggest changes.

Good luck!

**Review by zero2one — APPROVED** [review]

**Pull request closed without merging**